app/controllers/comments_controller.rb

```diff
@@ -11,11 +11,33 @@ def edit
   def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
-
     respond_to do |format|
       if @comment.save
+        reference = @comment.root.commentable
+
+        if @comment.parent
+          parent_id = @comment.parent_id
+          new_comment = @comment.parent.children.build
+          display_comment = false
+        else
+          parent_id = "r#{reference.id}"
```

Comment:
This doesn't seem like a good idea. I'd prefer to switch based on a class or some way of indicating the type of element you want to anchor the comment to.